### PR TITLE
Fix flaky unit test due to random data

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Fixes https://github.com/weaveworks/eksctl/issues/2105

### Approach

Underlying issue is due to https://github.com/bxcodec/faker/blob/1c6d6edb0123edda0348adc3ccb7e3480b9aaf07/faker.go#L488, which might cause the slice length as zero. I can enforce the length of slice with tag e.g. `faker:"len=1"`, but it's not looking good in struct. 

The default value https://github.com/bxcodec/faker/blob/1c6d6edb0123edda0348adc3ccb7e3480b9aaf07/faker.go#L29 is 100, so this test case will probably fail one in hundred times. We can also increase default values to reduce the odds, but it still fails some time :)

In short, I just add the check in case generated slice is empty.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by martina-if** [comment]
Hi @sayboras thanks for fixing this. I've seen this error before too. Let's get rid of this test and add a deterministic test instead with 2 or 3 profile selectors. What do you think?

**Comment by sayboras** [comment]
> Hi @sayboras thanks for fixing this. I've seen this error before too. Let's get rid of this test and add a deterministic test instead with 2 or 3 profile selectors. What do you think?

@martina-if it's the best option I can say. Let me remove such randomness in unit test. Maybe dependency _fake_ might not be required anymore.
